### PR TITLE
TGchat: Tweaks & Fixes

### PR DESCRIPTION
**Pull request by AyIong**
## About The Pull Request
1. Case Sensitive highlight has been fixed, it works now
2. Added `Clear Chat` button, clears the active tab. Code stolen from GoonStation closed PR
https://github.com/goonstation/goonstation/pull/16708
3. Added `Mute` button, it allows to switch off Unread counter on separate tabs
4. Cyrillic is supported by highlights again
5. Small appearance edits, unfortunately dropdowns are just as bad
6. Buttons when pressed no longer remain highlighted as if you were hovering over them, which was slightly misleading and generally looked like a bug (I'l revert it if you say so)

## Why It's Good For The Game
You can mute individual tabs so they don't clutter your eyes, clear chat if you need to and create Case Sensitive highlights, isn't that nice?

## Images of Changes

<details>
<summary>Case Sensitive</summary>

https://github.com/tgstation/tgstation/assets/69762909/1123049c-05e1-4110-82de-79c2f807453f

</details>
<details>
<summary>Clear Chat</summary>

https://github.com/tgstation/tgstation/assets/69762909/6ac91d08-2a22-435e-974d-7386ade77b77

</details>
<details>
<summary>Mute Button</summary>

https://github.com/tgstation/tgstation/assets/69762909/394dbe46-2e6f-4217-840c-d9a7ebaff566

</details>

## Changelog

:cl: Aylong
add: Added `Mute` button into `Chat Tabs` settings, it disables tab unread counter
add: Added `Clear chat` button into `General` settings, you can clear your dirty chat like you did it before TGchat
fix: Case-sensitive highlighting now works properly
/:cl:

### CODE REVIEW

**Comment by LemonInTheDark** [comment]
You need permission from the author of https://github.com/goonstation/goonstation/pull/16708 to relicense their code if you're using it

**Comment by Aylong220** [comment]
Oh... ok

File: tgui/packages/tgui-panel/chat/replaceInTextNode.js [hunk 1]
```diff
@@ -93,7 +93,7 @@ export const replaceInTextNode = (regex, words, createNode) => (node) => {
     for (let word of words) {
       // Capture if the word is at the beginning, end, middle,
       // or by itself in a message
-      wordRegexStr += `^${word}\\W|\\W${word}\\W|\\W${word}$|^${word}$`;
+      wordRegexStr += `^${word}\\s\\W|\\s\\W${word}\\s\\W|\\s\\W${word}$|^${word}\\s\\W$`;
```
Comment (optimumtact) [inline tgui/packages/tgui-panel/chat/replaceInTextNode.js:96]: is this the change to enable cyrillic in highlights?

Comment (Aylong220) [reply tgui/packages/tgui-panel/chat/replaceInTextNode.js:96]: Fixes Case Sensitive and gives ability to enable cyrillic highlights without crash

**Comment by ZeWaka** [comment]
> Hm ok. idk honestly style was the one to license his tgui stuff under mit so I'll let it be his problem. I think it's prob fine

it is, that code is MIT licensed - tg never gave permission to relicense tgui

**Comment by optimumtact** [comment]
@ZeWaka hey can you check htis

**Comment by CheffieGithub** [comment]
Yeah I was going to pr some of my changes back to mainline anyway

**Comment by AyIong** [comment]
> Yeah I was going to pr some of my changes back to mainline anyway

<3

**Pull request merged**